docs/plugins/plugin-script.asciidoc

```diff
@@ -0,0 +1,229 @@
+Elasticsearch version.
+
+[float]
+==== Community and non-core plugins
+
+Non-core plugins provided by Elasticsearch, or plugins provided by the
+community, can be installed from `download.elastic.co`, from Maven (Central
+and Sonatype), or from GitHub.  In this case, the command is as follows:
+
+[source,shell]
+-----------------------------------
+bin/plugin --install [org]/[user|component]/[version]
+-----------------------------------
+
+For instance, to install the https://github.com/lmenezes/elasticsearch-kopf[Kopf]
+plugin from GitHub, run one of the following commands:
+
+[source,shell]
+-----------------------------------
+bin/plugin --install lmenezes/elasticsearch-kopf <1>
+bin/plugin --install lmenezes/elasticsearch-kopf/1.x <2>
+-----------------------------------
+<1> Installs the latest version from GitHub.
+<2> Installs the 1.x version from GitHub.
+
+When installing from Maven Central/Sonatype, `[org]` should be replaced by
+the artifact `groupId`, and `[user|component]` by the `artifactId`.  For
+instance, to install the
+https://github.com/elastic/elasticsearch-mapper-attachments[mapper attachment]
+plugin from Sonatype, run:
+
+[source,shell]
+-----------------------------------
+bin/plugin --install org.elasticsearch/elasticsearch-mapper-attachments/2.6.0 <1>
+-----------------------------------
+<1> When installing from `download.elastic.co` or from Maven Central/Sonatype, the
+    version is required.
+
+[float]
+==== Custom URL or file system
+
+A plugin can also be downloaded directly from a custom location by specifying the URL:
+
+[source,shell]
+-----------------------------------
+bin/plugin --url [url] --install [plugin-name] <1>
+-----------------------------------
+<1> Both the URL and the plugin name must be specified.
+
+For instance, to install a plugin from your local file system, you could run:
+
+[source,shell]
+-----------------------------------
+bin/plugin --url file:///path/to/plugin.zip --install my_plugin
+-----------------------------------
+
+.Installation process
+*************************************
+
+The plugin installer tries to download a Zip file from a custom URL (if
+specified), from `download.elastic.co`, Maven Central, Sonatype, or GitHub.
+If successful, it unzips the plugin into the `plugins/` directory, usually
+located under `$ES_HOME`.
+
+If a plugin contains a `bin/` or `config/` directory, their contents are moved
+to the `$ES_HOME/bin` and `$ES_HOME/config` directories respectively.
+
+If any other non-JAR files are present and the plugin doesn't already contain
+a `_site/` directory, that directory is created and all non JAR files are moved
+into it.
+
+*************************************
+
+
+[[listing-removing]]
+=== Listing and Removing Installed Plugins
+
+[float]
+==== Listing plugins
+
+A list of the currently loaded plugins can be retrieved with the `--list` option:
+
+[source,shell]
+-----------------------------------
+bin/plugin --list
```

Comment:
Unfortunately current version of the plugin manager does not support `list` command but only `--list` or `-l`. @clintongormley do you want me to open a small PR adding support for `list` so we will be consistent with `install` and `remove`?